src/pages/Editor/Classes/UpdateClassModals/EditClassModal.tsx

```diff
@@ -0,0 +1,165 @@
+  item,
+  isOpen,
+  toggle,
+  fetchClasses,
+}: EditClassModalProps) => {
+  const {
+    register,
+    control,
+    setValue,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<CreateClassProps>({
+    defaultValues: {
+      classworks: item.classworks,
+    },
+    mode: 'onChange',
+  });
+  const { fields, append, remove } = useFieldArray({
+    name: 'classworks',
+    control,
+  });
+  useEffect(() => {
+    setValue('name', item.name);
+    setValue('description', item.description);
+  }, [item.name, item.description, setValue]);
+
+  const [errorMessages, setErrorMessages] = useState<ErrorMessageInterface[]>(
+    []
+  );
+
+  // ClassWork
+
+  const handleCancelModal = () => {
+    setValue('name', item.name);
+    setValue('description', item.description);
+    reset();
+    toggle();
+  };
+
+  const onSubmit = async (data: CreateClassProps) => {
+    try {
+      await editClass(data, item._id);
+      fetchClasses();
+      reset();
+      toggle();
+    } catch (error) {
+      setErrorMessages(error as ErrorMessageInterface[]);
+    }
+  };
+
+  return (
+    <Modal
+      titleText="Edit Class"
+      isOpen={isOpen}
+      onCloseModal={toggle}
+      primaryAction={
+        <PrimaryButton type="submit" form="EditClassForm">
+          Submit
+        </PrimaryButton>
+      }
+      secondaryAction={<Button onClick={handleCancelModal}>Cancel</Button>}
+      customStyles={{
+        content: {
+          height: '90vh',
+          width: '90vw',
+        },
+      }}
+    >
+      <StyledForm id="EditClassForm" onSubmit={handleSubmit(onSubmit)}>
+        <InputField htmlFor="name">
+          <span>Class Title</span>
+          <Input
+            type="text"
+            id="name"
+            placeholder="Enter Class Title"
+            {...register('name', classValidationRules.name)}
+          />
+        </InputField>
+        <InputField htmlFor="description">
+          <span>Class Description</span>
+          <TextArea
+            id="description"
+            placeholder="Enter Class description"
+            {...register('description', classValidationRules.description)}
+          />
+        </InputField>
+
+        {fields.map((field, index) => {
+          return (
+            <Classworks key={field.id}>
+              <Input
+                placeholder="name"
+                {...register(`classworks.${index}.name` as const, {
+                  required: true,
+                })}
+                defaultValue={field.name}
+              />
+              <Input
+                placeholder="description"
+                {...register(`classworks.${index}.description` as const, {
+                  required: true,
+                })}
+                defaultValue={field.description}
+              />
+
+              <TrashIcon type="button" onClick={() => remove(index)} />
+            </Classworks>
+          );
+        })}
+        <PrimaryButton
+          type="button"
```

Comment:
I think we don't need this since `PrimaryButton` is already a `button` as the base?